libs/entitlement/src/main/java/org/elasticsearch/entitlement/bootstrap/EntitlementBootstrap.java

```diff
@@ -101,19 +101,49 @@ private static String findAgentJar() {
     }
 
     /**
+     * Attempt a few sensitive operations to ensure that some are permitted and some are forbidden.
+     *
      * @throws IllegalStateException if the entitlements system can't prevent an unauthorized action of our choosing
      */
     private static void selfTest() {
-        var pb = new ProcessBuilder(""); // The command doesn't matter; it doesn't even need to exist
+        ensureCannotStartProcess();
+        ensureCanCreateTempFile();
+    }
+
+    private static void ensureCannotStartProcess() {
         try {
-            pb.start();
+            // The command doesn't matter; it doesn't even need to exist
+            new ProcessBuilder("").start();
         } catch (NotEntitledException e) {
-            logger.debug("Success: Entitlement protection correctly prevented disallowed action");
+            logger.debug("Success: Entitlement protection correctly prevented process creation");
             return;
         } catch (IOException e) {
             throw new IllegalStateException("Failed entitlement protection self-test", e);
         }
-        throw new IllegalStateException("Failed entitlement protection self-test");
+        throw new IllegalStateException("Entitlement protection self-test was incorrectly permitted");
+    }
+
+    /**
+     * Originally {@code Security.selfTest}.
+     */
+    @SuppressForbidden(reason = "accesses jvm default tempdir as a self-test")
+    private static void ensureCanCreateTempFile() {
+        try {
+            Path p = Files.createTempFile(null, null);
+            p.toFile().deleteOnExit();
```

Comment:
(The original from `Security.selfTest` didn't do this. We could remove it if it's a problem.)

Reply:
I think this is a good idea

Reply:
Thanks! My only hesitation is that the list of files to delete is arguably JVM-wide state that would need its own entitlement. But if this ever becomes awkward at all, we can simply delete this `deleteOnExit` call and get on with our lives.